### PR TITLE
Use loops to spawn needed golems

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

**Review by qndel — APPROVED** [review]
👍

File: Source/monster.cpp [hunk 1]
```diff
@@ -3831,10 +3831,8 @@ void monster_some_crypt()
 void InitMonsters()
 {
 	if (!setlevel) {
-		AddMonster(GolemHoldingCell, DIR_S, 0, false);
-		AddMonster(GolemHoldingCell, DIR_S, 0, false);
-		AddMonster(GolemHoldingCell, DIR_S, 0, false);
-		AddMonster(GolemHoldingCell, DIR_S, 0, false);
+		for (int j = 0; j < MAX_PLRS; j++)
```
Comment (julealgon) [inline Source/monster.cpp:3834]: Can't we use something like `for (auto _ : players)` here and avoid the extra dependency with `MAX_PLRS`?

File: Source/monster.cpp [hunk 2]
```diff
@@ -3897,10 +3895,9 @@ void InitMonsters()
 void SetMapMonsters(const uint16_t *dunData, Point startPosition)
 {
 	AddMonsterType(MT_GOLEM, PLACE_SPECIAL);
-	AddMonster(GolemHoldingCell, DIR_S, 0, false);
-	AddMonster(GolemHoldingCell, DIR_S, 0, false);
-	AddMonster(GolemHoldingCell, DIR_S, 0, false);
-	AddMonster(GolemHoldingCell, DIR_S, 0, false);
+	for (int k = 0; k < MAX_PLRS; k++)
```
Comment (julealgon) [inline Source/monster.cpp:3898]: The use of `j` and then `k` to represent the same thing in the same commit looks weird to me.

File: Source/monster.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/monster.cpp:3834]: That seams really missleading

File: Source/monster.cpp (hunk 2 above)
Comment (AJenbo) [reply Source/monster.cpp]: Fixed

File: Source/monster.cpp (hunk 1 above)
Comment (julealgon) [reply Source/monster.cpp]: I don't know. I think the `MAX_PLRS` macro should not exist.

Comment (AJenbo) [reply Source/monster.cpp]: Not disagreeing, but that is a refactor for a later day :) Just pulled this out of another PR so that it could get closed.

**Pull request merged**